example/src/main/tut/data/Kleisli.md

```diff
@@ -0,0 +1,137 @@
+
+
+## Basics
+
+While this may seem like a complicated topic, in practice, the notion of a `Kleisli` is simple. Any function `A => M[B]` is a `Kleisli`. The `Kleisli` data object provides a succinct newtype wrapper for easy costless conversion between `Kleisli` and its corresponding function type via two special functions, `wrapKleisli`, and `runKleisli`:
+
+```tut:silent
+import scalaz._
+import Scalaz._
+import data._ // this gets us our Kleisli object
+
+val f: Int => List[Boolean] = i => List(i < 2)
+
+val k: Kleisli[List, Int, Boolean] = Kleisli.wrapKleisli(f)
+val ff: Int => List[Boolean] = Kleisli.runKleisli(k)
+
+```
+
+Note the following:
+
+```
+Kleisli.wrapKleisli(Kleisli.runKleisli(k)) === k
+
+// And
+
+Kleisli.runKleisli(Kleisli.wrapKleisli(f)) === f
+```
+
+Additionally, we provide the following utility allowing one to translate `Kleisli`s from `F[_]` to `G[_]` according to some natural transformation (`~>`) for convenience:
+
+```
+  def hoist[F[_], G[_], A, B](k: Kleisli[F, A, B])(η: F ~> G): Kleisli[G, A, B]
+```
+
+
+As with any function, the most important operations we can do, aside from application, is composition. Note that this poses a problem for `Kleisli`. How do we compose `A => M[B]` and `B => M[C]`? This is where the magic begins. It turns out that when M is a `Monad`, then `Kleisli`s compose via M's `flatMap`:
+
+```
+
+  def compose[F[_], A, B, C](
+    j: Kleisli[F, B, C],
+    k: Kleisli[F, A, B]
+  )(implicit M: Monad[F]): Kleisli[F, A, C] =
+    a => M.flatMap(k(a))(j)
+```
+
+With a little imagination, we can see that the `Kleisli` construction not only forms a new Category, but when M is a `Monad`, `Kleisli`s form a `Monad` as well!
+
+## Functions
+
+We provide the following functions in addition to `wrapKleisli`, `runKleisli`, and `compose` for `Kleisli`:
+
+```
+
+    // turn F into G via some natural transformation
+    def hoist[G[_]](η: F ~> G): Kleisli[G, A, B]
+
+    // analogous to `andThen` for functions
+    def andThen[C](
+      j: Kleisli[F, B, C]
+    )(implicit M: Monad[F]): Kleisli[F, A, C]
+
+    // analogous to `compose` for functions
+    def compose[E](
+      j: Kleisli[F, E, A]
+    )(implicit M: Monad[F]): Kleisli[F, E, B]
+
+    // symbolic version of `andThen`
+    def >=>[C](j: Kleisli[F, B, C])(implicit M: Monad[F]): Kleisli[F, A, C]
+
+    // symbolic version of `compose`
+    def <=<[E](j: Kleisli[F, E, A])(implicit M: Monad[F]): Kleisli[F, E, B]
+
+    // reverse version of `>>=`
+    def =<<(fa: F[A])(implicit M: Monad[F]): F[B]
+
+    // An alias for `andThen`
+    def >>>[C](j: Kleisli[F, B, C])(implicit M: Monad[F]): Kleisli[F, A, C]
+
+    // Split the input between the two argument arrows and combine their output. Note that this is in general not a functor.
+    def ***[C, D](j: Kleisli[F, C, D])(
+      implicit S: Strong[Kleisli[F, ?, ?]],
```

Comment:
What does "this" refer to? I.e., what is not a functor?

Reply:
Left in there from extra note-taking. It doesn't apply to `Kleisli`, but `Arrows` in general